frontend/src/components/form/password-field.tsx

```diff
@@ -0,0 +1,33 @@
+import { ViewIcon, ViewOffIcon } from '@chakra-ui/icons';
+import { Button, FormControl, FormLabel, Input, InputGroup, InputRightElement, useBoolean } from '@redpanda-data/ui';
+import { ErrorInfoField } from './error-info-field';
+import { useFieldContext } from './form-hook-contexts';
+
+interface PasswordFieldProps {
+  label: string;
+}
+
+export const PasswordField = ({ label }: PasswordFieldProps) => {
+  const field = useFieldContext<string>();
+  const [showValue, setShowValue] = useBoolean(false);
+
+  return (
+    <FormControl isInvalid={!!field.state.meta.errors?.length}>
```

Comment:
nit: I though we had a PasswordInput component in the existing UI lib. Since we're using components from that library withing the file, could we use it as well?